### PR TITLE
net: nrf_provisioning: Add new API to set provisioning interval 

### PR DESCRIPTION
**Pull request by juhaylinen**
Add new API and shell command to set provisioning interval.
Change DTLS CID from enabled to supported .
Update documentation and change log.

### CODE REVIEW

File: doc/nrf/libraries/networking/nrf_provisioning.rst [hunk 1]
```diff
@@ -156,10 +156,11 @@ The feature is enabled by selecting :kconfig:option:`CONFIG_NRF_PROVISIONING_SHE
...
    Subcommands:
-     init   :Start the client
-     now    :Do provisioning now
-     token  :Get the attestation token
-     uuid   :Get device UUID
+     init      :Start the client
+     now       :Do provisioning now
+     token     :Get the attestation token
+     uuid      :Get device UUID
+     interval  : Set provisioning interval
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_provisioning.rst:163]:
```suggestion
     init: Start the client
     now: Do provisioning now
     token: Get the attestation token
     uuid: Get device UUID
     interval: Set provisioning interval
```

File: subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c [hunk 2]
```diff
@@ -512,6 +512,12 @@ static void commit_latest_cmd_id(void)
 	}
 }
 
+void nrf_provisioning_set_interval(int interval)
+{
+	nxt_provisioning = interval;
+	LOG_DBG("Provisioning interval set to %d", interval);
```
Comment (SeppoTakalo) [inline subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c:518]: Don't we need to re-schedule here?

File: include/net/nrf_provisioning.h [hunk 3]
```diff
@@ -106,6 +106,13 @@ int nrf_provisioning_init(struct nrf_provisioning_mm_change *mmode,
  */
 int nrf_provisioning_trigger_manually(void);
 
+/**
+ * @brief Sets provisioning interval.
```
Comment (peknis) [inline include/net/nrf_provisioning.h:110]:
```suggestion
 * @brief Set provisioning interval.
```

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -342,7 +342,9 @@ Modem libraries
 Libraries for networking
 ------------------------
 
-|no_changes_yet_note|
+* :ref:`lib_nrf_provisioning` library:
+
+   * Added the :c:func:`nrf_provisioning_set_interval` API function to set the interval between provisioning attempts.
```
Comment (peknis) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:347]:
```suggestion
   * Added the :c:func:`nrf_provisioning_set_interval` function to set the interval between provisioning attempts.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Review by SeppoTakalo — APPROVED** [review]

File: subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c (hunk 2 above)
Comment (juhaylinen) [reply subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c:519]: Updated the function to reschedule and now the interval is saved to storage.

**Comment by juhaylinen** [comment]
@peknis I have fixed the documentation. Please check.

**Review by peknis — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

File: subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c [hunk 5]
```diff
@@ -512,6 +514,42 @@ static void commit_latest_cmd_id(void)
...
+
+		ret = snprintf(time_str, sizeof(time_str), "%d", interval);
+		if (ret < 0) {
+			LOG_ERR("Unable to convert interval to string");
+			return;
+		}
+
+		ret = settings_save_one(SETTINGS_STORAGE_PREFIX "/interval-sec",
+			time_str, strlen(time_str) + 1);
+		if (ret) {
```
Comment (SeppoTakalo) [inline subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c:545]: Do you know why we store the time as a string?

Settings storage should be able to store the `int` value.

Not asking to change here, but if you have time, that could be refactored.

Comment (juhaylinen) [reply subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c:545]: Probably because config commands contain key-value pairs in string format.

**Pull request merged**